GitUI/UserControls/RevisionGrid/RevisionGridControl.Designer.cs

```diff
@@ -189,6 +193,22 @@ private void InitializeComponent()
             this.bisectSeparator.Name = "bisectSeparator";
             this.bisectSeparator.Size = new System.Drawing.Size(261, 6);
             // 
+            // popStashToolStripMenuItem
+            // 
+            this.popStashToolStripMenuItem.Image = global::GitUI.Properties.Images.Stash;
+            this.popStashToolStripMenuItem.Name = "popStashToolStripMenuItem";
+            this.popStashToolStripMenuItem.Size = new System.Drawing.Size(264, 24);
+            this.popStashToolStripMenuItem.Text = "Pop &stash...";
+            this.popStashToolStripMenuItem.Click += new System.EventHandler(this.PopStashToolStripMenuItemClick);
+            // 
+            // dropStashToolStripMenuItem
+            // 
+            this.dropStashToolStripMenuItem.Image = global::GitUI.Properties.Images.Stash;
+            this.dropStashToolStripMenuItem.Name = "dropStashToolStripMenuItem";
+            this.dropStashToolStripMenuItem.Size = new System.Drawing.Size(264, 24);
+            this.dropStashToolStripMenuItem.Text = "&Drop stash...";
```

Comment:
These dots imply there was a confirmation dialog. But it is missing. Add the optional confirmation, please. We already have a regarding AppSetting (which I do not turn off).
Dropped stashes are hard to recover. They do not appear in the RefLog.
After `recovering` the `lost object`, the commit is filtered out somehow as long as RefLog references are shown.

Reply:
OK
Making a temporary change to hide the option for now

Reply:
> Making a temporary change to hide the option for now

OK. We can activate it in a follow-up - as well as add `Apply stash`.

Reply:
"stash apply" is basically the same command as git-cherry-pick so it is not really needed. stash apply can be applied to any commit.

But as the menu is only for stash I guess it is not much in the way.

To be clear: I will remove the stash-drop (unless I get around to create a confirmation) before merging this if approved.

Reply:
> I will remove the stash-drop (unless I get around to create a confirmation) before merging this if approved.

It has already been implemented in FormStash - with optional confirmation.
If you omit this menu item, I will add it myself.

> "stash apply" is basically the same command as git-cherry-pick.

Is this also true for complex stashes, e.g. with untracked files?

Reply:
I added _Apply stash_ as well as a confirmation for drop stash
(Same keyboard accelerator for stash-apply as for cherry-pick)

> Is this also true for complex stashes, e.g. with untracked files?

The dialog lists the untracked commit but do add the files. (index commit files are included, but they are included in the main commit already). Could be an issue in the dialog, not investigated (do not plan to).

I just did not see the usefulness of appy stash myself, but if you ask for it, other may do that too.
Did not want to add too many bells and whistles before I know if it will be accepted.